### PR TITLE
Allow building the documentation with gi-docgen and gtk-doc

### PR DESCRIPTION
**Pull request by hughsie**
Until gi-docgen is declared stable support both of them.
This effectively means that hand builds and CI builds will use
gi-docgen, but distro builds use gtk-doc-tools.

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [X] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
Alternate idea. It seems a bit less magic this way...

File: meson.build [hunk 1]
```diff
@@ -535,10 +535,16 @@ endif
...
-  gtkdocscan = find_program('gtkdoc-scan', required : true)
-  subdir('docs')
+if get_option('docs') == 'gtkdoc'
+  gtkdocscan = find_program('gtkdoc-scan')
+elif get_option('docs') == 'docgen'
+  gidocgen_dep = dependency('gi-docgen',
+    version: '>= 2021.1',
+    fallback: ['gi-docgen', 'dummy_dep'],
+  )
+  gidocgen = find_program('gi-docgen')
```
Comment (superm1) [inline meson.build:545]: I think you need a third option for anything else to error out, "Unknown doc type" or something?

Comment (superm1) [reply meson.build:545]: Or does meson enforce the choices?

**Review by superm1 — APPROVED** [review]
LGTM

File: meson.build (hunk 1 above)
Comment (hughsie) [reply meson.build:545]: meson configure -Ddocs=XXXX
    ERROR: Value "XXXX" (of type "string") for combo option "developer documentation type" is not one of the choices. Possible choices are (as string): "none", "gtkdoc", "docgen".

**Comment by mutte-79** [comment]
Would I find the information you need if I get a printout from the memory i EFI?
I'll do the code building too. Maybe I need that as a little lesson.
But boy was I angry a while ago. Oh my god.

**Comment by mutte-79** [comment]
Atm. I'm using EFI v2.13 but the EFI SHELL only runs at v1
It's supposed to be 4.16, I think.
That's why I have been collecting Grub and that nice spanking new bootx64.EFI
I hope I am getting at least 1% of this right. I am balls tired.

**Comment by mutte-79** [comment]
Hi, I took these for Linux Hardware and thought you might have some use for them too.
It's a probe with almost everything turned on in my BIOS. and a pretty damn long list from CPUID.


Med vänliga hälsningar
With kind regards

Mattias Högfjäll

________________________________

**Comment by mutte-79** [comment]
Hi there.

I'm probably going to reinstall Ubuntu in a near future. This installation has been extremely weird and unstable. I don't know if your program has made any progress but if there's any files you would like me to bring along let me know.
What is it the program wants?
It keeps repeating:

03:06:08:0975 FuEngine emmc failed to change udev device /sys/devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:0/0:0:0:0/block/sda/sda6: cannot ensure ID: FuUdevDevice:

I started gpart yesterday and that seemed to have made the program happy for a little while but then it continued repeating.
I saw some of the files mentioning Dell. That's just to have something to fill in or what? There're no Dell parts in this computer.




Med vänliga hälsningar
With kind regards

Mattias Högfjäll
________________________________

**Comment by mutte-79** [comment]
Hi there.

I Don't know if you got my last mail but I upgraded from Kubuntu 20.10 to 21.04 and something went sour with the Fwupd install at first. I saw later on that Ubuntu continued installing stuff that had failed.
But now I get warnings from SMART all the time, that my disks are braking. 2/4. The numbers seem way off too. I don't know if Fwupd is involved in that but if it is, would you be so kind and take a look?
I can't post anything from my logs unfortenately because I have a small window compositor meltdown at the moment so only like 1% of my windows gets any content written in them.
As soon as I get some filling in my windows I'll send some examples.




Med vänliga hälsningar
With kind regards

Mattias Högfjäll
________________________________

**Comment by mutte-79** [comment]
Hi again.
The message "fwupd: /dev/mem,kmem,port is restricted; see man kernel_lockdown.7" is repeating but I have so many other issues that I'm going to reinstall Kubuntu. Somehow the upgrade didn't go too well and my hysterical testing of all the little programs I can get hold of I dont se the point of trying to fix it.




Med vänliga hälsningar
With kind regards

Mattias Högfjäll
________________________________

**Comment by hughsie** [comment]
@mutte-79 this is a issue tracker for specific issues to do with the fwupd project. We can't help you with these issues I'm afraid.

**Pull request merged**